src/printer/dagification_visitor.cpp

```diff
@@ -59,7 +59,7 @@ bool DagificationVisitor::alreadyVisited(TNode current, TNode parent) {
     // search for variables that start with the let prefix
     std::unordered_set<TNode, TNodeHashFunction> vs;
     expr::getVariables(current, vs);
-    for (const TNode v : vs)
+    for (const TNode& v : vs)
```

Comment:
So actually, after having thought about this some more (and written [this](https://github.com/CVC4/CVC4/wiki/Developer-Guide#dealing-with-expressions-nodes-and-tnodes)) I might even argue for only having `const TNode` here... :smile:

Reply:
I see your point but I think that we anyway make an exception for range-based for loops with `const Node&`, so `const TNode&` IMHO makes sense for the sake of uniformity (and to shut up the compiler :) ).

Reply:
I guess I agree. The only argument for `const TNode` would be improved cache locality, and I'd hope that this is not a big issue for a loop...